go/mysql/sql_error.go

```diff
@@ -87,6 +87,9 @@ func NewSQLErrorFromError(err error) error {
 		return nil
 	}
 
+	// in case this error is wrapped by vterrors, extract the inner most error
+	err = vterrors.UnwrapAll(err)
```

Comment:
This is breaking an existing test. So we may be expecting the current functionality elsewhere. Tagging @GuptaManan100 since the test was added fairly recently by him.

```
  server_flaky_test.go:1290: 
2023-02-15T10:11:35.1823236Z         	Error Trace:	/home/runner/work/vitess/vitess/go/mysql/server_flaky_test.go:1290
2023-02-15T10:11:35.1823826Z         	Error:      	"Row count exceeded 10000 (errno 1317) (sqlstate 70100) during query: error" does not contain "wrapped: Row count exceeded 10000"
2023-02-15T10:11:35.1824147Z         	Test:       	TestErrorCodes/wrapped:_Row_count_exceeded_10000
```

Reply:
FWIW I intend to do more work on `NewSQLErrorFromError` in a separate PR. At any case happy to accommodate.

Reply:
Sweeeet! Thank-you!